cms/djangoapps/contentstore/views/videos.py

```diff
@@ -285,33 +286,46 @@ def convert_video_status(video):
     return status
 
 
-def _get_videos(course):
+def _get_videos(course, request):
     """
     Retrieves the list of videos from VAL corresponding to this course.
     """
-    videos = list(get_videos_for_course(course.id, VideoSortField.created, SortDirection.desc))
+    params = {
+        "page": request.GET.get("page", 1),
+        "page_size": request.GET.get("page_size", 20),
+        "paginated": True,
+        "sort_field": VideoSortField[request.GET.get("sort_field", "created")],
+        "sort_dir": SortDirection[request.GET.get("sort_dir", "asc")]
+    }
 
+    videos_data = get_videos_for_course(course.id, **params)
+    videos = videos_data["results"]
     # convert VAL's status to studio's Video Upload feature status.
     for video in videos:
```

Comment:
I would loop on videos_data["results"] directly instead of assigning it to `videos`.